### DISCUSSION TITLE
求助client hello标准

### DISCUSSION THREAD

**Discussion by ghost** [URenko/Accesser#201] [Ideas]
见此https://github.com/monthlyliaoyuan/Unofficial_Z_Access/issues/12

**Comment by URenko** [comment]
？
没明白你在做什么以及包的来源

client hello (TLS 1.2) 标准在 https://datatracker.ietf.org/doc/html/rfc5246#section-7.4.1.2

**Reply by ghost** [reply]
> 在

这是一个本地代理，会拦截tcp连接建立后的第一个包(https下就是clinet hello)然后分片发出。

理论上，设计明文sni,gfw会组装分片，会被rst。但是实际测试中，平均七八次连接会有一次成功连上。
我在想这些包有啥区别，为啥有点rst有点的不会。
似乎区别就是那几个没有被rst的包sni出现在很后面，跟两个/??和几个/x00就结束了

**Reply by ghost** [reply]
代理跑在传输层，所以我也不知道应用层的chromium是怎么构造clinethello的。

**Reply by URenko** [reply]
建议开个 wireshark 看吧，读二进制太累了

**Comment by ghost** [comment]
代码在https://github.com/maoist2009/TLSfragment

**Comment by URenko** [comment]
我记得 fragment 有不少人做过，可以调查下现有研究

**Reply by ghost** [reply]
伊朗的研究，我改的gfw_knocker的项目，主要是确保sni被分片。但是疑似说中国墙会组装碎片，在中国是无效的。
但是事实是总会有一定纪几率连接成功。和torproject的连接甚至一直成功。

**Comment by URenko** [comment]
https://github.com/XTLS/Xray-core/pull/2021
https://github.com/XTLS/Xray-core/pull/2131
https://github.com/net4people/bbs/issues/308
（俄罗斯）https://github.com/ValdikSS/GoodbyeDPI

**Comment by URenko** [comment]
IP 地址、路由的一致性也是需要检查的点

**Reply by ghost** [reply]
ip是写死的
路由不会查。

**Comment by ghost** [comment]
我电脑没管理admin,用不了wireshark.
真是可笑

________________________________
发件人: URenko ***@***.***>
发送时间: 2024年10月4日 16:57
收件人: URenko/Accesser ***@***.***>
抄送: loueisun ***@***.***>; Author ***@***.***>
主题: Re: [URenko/Accesser] 求助clinet hello标准 (Discussion #201)


建议开个 wireshark 看吧，读二进制太累了

―

**Reply by jonm58** [reply]
> 我电脑没管理admin,用不了wireshark. 真是可笑

你不会自己创建一个admin账户或者修改账户的账户组?

**Reply by ghost** [reply]
> > 我电脑没管理admin,用不了wireshark. 真是可笑
> 
> 你不会自己创建一个admin账户或者修改账户的账户组?
> 
> 

我爸没给admin，我也懒得问他要。

**Reply by URenko** [reply]
那你怎么抓的包，介入网络栈需要管理员权限。

**Reply by jonm58** [reply]
你是不是开个VM，在VM那里抓的包?

**Reply by ghost** [reply]
浏览器代理 switch peoxy omega或者命令行--proxy-server

**Reply by URenko** [reply]
wireshark 单用于分析不用于抓包（不安装 winpcap / npcap）应该不需要管理员权限

你用代理的方式抓得包后保存为 pcap （ https://github.com/muzuiget/mitmpcap 或 https://www.netresec.com/?page=PolarProxy 可能是现成的工具）放 wireshark 里查看就行了。

**Reply by ghost** [reply]
没管理进不到数据链路曾层，好像抓不到传输层的ack,rst之类的消息。

**Reply by Lanius-collaris** [reply]
你有可以启动操作系统的U盘吗？有的话从那个U盘启动。
传说EtchDroid USB writer可以在没有root的情况下创建可以启动Linux的U盘。
https://f-droid.org/en/packages/eu.depau.etchdroid/
不知道EtchDroid USB writer在你的手机上能不能用。

**Reply by ghost** [reply]
谢谢，但我比较忙，不一定有空刷一个linux.而且我电脑性能不咋地。
